app/assets/javascripts/Components/course_summaries_table.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import {render} from 'react-dom';
+
+import ReactTable from 'react-table';
+
+
+class CourseSummaryTable extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      data: [],
+      grade_columns: [],
```

Comment:
This and the next two field names aren't good. This particular one is really about assignments only. The next two should be pluralized.

But also it's better to just store one attribute that has *all* the columns anyway. There's no reason to store the three columns separately.